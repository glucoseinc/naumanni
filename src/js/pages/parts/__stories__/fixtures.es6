import {Account} from 'src/models'


const accountA = new Account({
  'username': 'kyounosuke1218',
  'acct': 'kyounosuke1218@pawoo.net',
  'display_name': '塚原藍森林✅Pawoo.net垢',
  'locked': false,
  'created_at': '2017-04-14T15:45:20.289Z',
  'followers_count': 281,
  'following_count': 612,
  'statuses_count': 295,
  'oauth_authentications': [{
    'uid': '19497031',
    'provider': 'pixiv',
  }],
  'note': '<p>「同人サークルツカラジッ！」とかいう団体で番組作ってます。<br /><a href=\"https://pawoo.net/tags/%E6%9D%B1%E5%8C%97\" class=\"mention hashtag\">#<span>東北</span></a> <a href=\"https://pawoo.net/tags/%E9%9D%92%E6%A3%AE\" class=\"mention hashtag\">#<span>青森</span></a> <a href=\"https://pawoo.net/tags/%E5%B2%A9%E6%89%8B\" class=\"mention hashtag\">#<span>岩手</span></a> <a href=\"https://pawoo.net/tags/%E9%89%84%E9%81%93\" class=\"mention hashtag\">#<span>鉄道</span></a> <a href=\"https://pawoo.net/tags/%E3%82%A2%E3%83%8B%E3%83%A1\" class=\"mention hashtag\">#<span>アニメ</span></a> <a href=\"https://pawoo.net/tags/%E3%82%AC%E3%82%B8%E3%82%A7%E3%83%83%E3%83%88\" class=\"mention hashtag\">#<span>ガジェット</span></a> <a href=\"https://pawoo.net/tags/%E3%83%80%E3%83%B3%E3%82%A8%E3%83%9C\" class=\"mention hashtag\">#<span>ダンエボ</span></a> <a href=\"https://pawoo.net/tags/%E3%82%A2%E3%82%A4%E3%83%89%E3%83%AB\" class=\"mention hashtag\">#<span>アイドル</span></a> <a href=\"https://pawoo.net/tags/%E9%9F%B3%E3%82%B2%E3%83%BC\" class=\"mention hashtag\">#<span>音ゲー</span></a><br /> <a href=\"https://twitter.com/kyounosuke1218\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">https://</span><span class=\"\">twitter.com/kyounosuke1218</span><span class=\"invisible\"></span></a><br /><a href=\"https://mstdn.jp/web/accounts/5\" rel=\"nofollow noopener\" target=\"_blank\"><span class=\"invisible\">https://</span><span class=\"\">mstdn.jp/web/accounts/5</span><span class=\"invisible\"></span></a><br /><script>alert(\"abracadabra\")</script></p>',
  'url': 'https://pawoo.net/@kyounosuke1218',
  'avatar': 'https://img.pawoo.net/accounts/avatars/000/009/229/original/932077979e1f0f59.png',
  'avatar_static': 'https://img.pawoo.net/accounts/avatars/000/009/229/original/932077979e1f0f59.png',
  'header': 'https://img.pawoo.net/accounts/headers/000/009/229/original/287c62eddef5e0f4.png',
  'header_static': 'https://img.pawoo.net/accounts/headers/000/009/229/original/287c62eddef5e0f4.png',
  'id_by_host': {
    'pawoo.net': 9229,
  },
})

export {accountA}
