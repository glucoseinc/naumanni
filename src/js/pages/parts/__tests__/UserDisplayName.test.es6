import React from 'react'
import {shallow} from 'enzyme'
import {spy} from 'sinon'
import renderer from 'react-test-renderer'


import {UserDisplayName} from '../UserParts'
import {mockAccount} from './fixtures'


const takeSnapshot = (component) => {
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
}

describe('UserDisplayName', () => {
  let component
  let wrapper

  describe('default', () => {
    beforeAll(() => {
      component = <UserDisplayName account={mockAccount} />
      wrapper = shallow(component)
    })
    it('has user-displayName class', () => {
      expect(wrapper.find('.user-displayName').length).toBe(1)
    })
    it('contains UserLink', () => {
      expect(wrapper.text()).toBe('<UserLink />')
    })
    it('match snapshot', () => {
      takeSnapshot(component)
    })
  })

  describe('onClick', () => {
    let onClickSpy = spy()

    beforeAll(() => {
      component = <UserDisplayName account={mockAccount} onClick={onClickSpy} />
      wrapper = shallow(component)
    })
    it('can be handled', () => {
      const a = wrapper.find('UserLink')
      a.simulate('click')
      expect(onClickSpy.calledOnce).toBeTruthy()
    })
  })
})
