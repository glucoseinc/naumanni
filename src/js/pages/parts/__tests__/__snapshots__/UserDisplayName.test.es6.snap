// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplayName default match snapshot 1`] = `
<a
  className="user-displayName"
  href={null}
  target="_blank"
>
  Shin Adachi
</a>
`;
